client/shared/src/main/scala/org/http4s/client/Client.scala

```diff
@@ -166,8 +166,8 @@ trait Client[F[_]] {
 
   /** Translates the effect type of this client from F to G
     */
-  def translate[G[_]: Async](fk: F ~> G)(gK: G ~> F)(implicit
-      F: MonadCancel[F, Throwable]): Client[G] =
+  def translate[G[_]: MonadCancelThrow](fk: F ~> G)(gK: G ~> F)(implicit
+      F: MonadCancelThrow[F]): Client[G] =
```

Comment:
Oh, sorry, actually this change is not binary compatible. You cannot change `Async` to `MonadCancelThrow` binary compatibly. Unfortunately this is not a Java-only problem.

We may be able to work around it in 0.23 by having two versions of this method, but not sure. At the very least we can name the other one `mapK` or something ...

Reply:
> You cannot change Async to MonadCancelThrow binary compatibly. Unfortunately this is not a Java-only problem.

Wait, why is this a problem in Scala?

Reply:
It's because binary compatibility doesn't care about the inheritance structure between `Async` and `MonadCancel`. They are different. This is a JVM level issue.

Reply:
Oh does that mean https://github.com/http4s/http4s/pull/5071 was bad and wrong too?

Reply:
Yikes, I think it was. Two issues got conflated there. The static-forwarder problem is Java only, but any signature change applies to Scala and Java (and any JVM language).

Reply:
In general, if MiMa complains, it is always right i.e. that change is not binary compatible. When we do exclusions we are effectively declaring that nobody on the outside is using that method (either b/c it's private/internal, or because it's only used by Java, and nobody uses http4s on Java, right?).

Reply:
> Yikes, I think it was. Two issues got conflated there. The static-forwarder problem is Java only, but any signature change applies to Scala and Java (and any JVM language).

Oops 😨 what's the fix, adding a package-private overload with the previous signature?

Reply:
Yes, that should do it. Java is still out in the cold but I think that was the original intention :)

Reply:
I think the package-private overload fixes this in classes, but not objects.  Objects still have that static forwarder problem, which we've waived a couple times for things that were causing clear and present problems, Java users be damned.

I'm +1 on this in 0.23 if it can be made MiMa clean, or if it can be reduced to a static forwarder and is causing real-world pain now.

Reply:
It looks like the package-private overload will work here. Do you want me to push it up on this branch, or should I push up a rebased version targeting series/0.23?

Reply:
Let's try targeting 0.23 if you don't mind :)